x-pack/plugins/actions/server/lib/get_execution_log_aggregation.ts

```diff
@@ -0,0 +1,446 @@
+                buckets_path: {
+                  count: 'actionExecution._count',
+                },
+                script: {
+                  source: 'params.count > 0',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+};
+
+export function getExecutionLogAggregation({
+  filter,
+  page,
+  perPage,
+  sort,
+}: IExecutionLogAggOptions) {
+  // Check if valid sort fields
+  const sortFields = flatMap(sort as estypes.SortCombinations[], (s) => Object.keys(s));
+  for (const field of sortFields) {
+    if (!Object.keys(ExecutionLogSortFields).includes(field)) {
+      throw Boom.badRequest(
+        `Invalid sort field "${field}" - must be one of [${Object.keys(ExecutionLogSortFields).join(
+          ','
+        )}]`
+      );
+    }
+  }
+
+  // Check if valid page value
+  if (page <= 0) {
+    throw Boom.badRequest(`Invalid page field "${page}" - must be greater than 0`);
+  }
+
+  // Check if valid page value
+  if (perPage <= 0) {
+    throw Boom.badRequest(`Invalid perPage field "${perPage}" - must be greater than 0`);
+  }
+
+  const dslFilterQuery: estypes.QueryDslBoolQuery['filter'] = buildDslFilterQuery(filter);
+
+  return {
+    executionLogAgg: {
+      ...(dslFilterQuery ? { filter: { bool: { filter: dslFilterQuery } } } : {}),
+      aggs: {
+        // Get total number of executions
+        executionUuidCardinality: {
+          filter: {
+            bool: {
+              must: [
+                {
+                  match: {
+                    [ACTION_FIELD]: 'execute',
+                  },
+                },
+              ],
+            },
+          },
+          aggs: {
+            executionUuidCardinality: {
+              cardinality: {
+                field: EXECUTION_UUID_FIELD,
+              },
+            },
+          },
+        },
+        executionUuid: {
+          // Bucket by execution UUID
+          terms: {
+            field: EXECUTION_UUID_FIELD,
+            size: DEFAULT_MAX_BUCKETS_LIMIT,
+            order: formatSortForTermSort(sort),
+          },
+          aggs: {
+            // Bucket sort to allow paging through executions
+            executionUuidSorted: {
+              bucket_sort: {
+                sort: formatSortForBucketSort(sort),
+                from: (page - 1) * perPage,
+                size: perPage,
+                gap_policy: 'insert_zeros' as estypes.AggregationsGapPolicy,
+              },
+            },
+            // Filter by action execute doc and get information from this event
+            actionExecution: {
+              filter: {
+                bool: {
+                  must: [
+                    {
+                      match: {
+                        [ACTION_FIELD]: 'execute',
+                      },
+                    },
+                  ],
+                },
+              },
+              aggs: {
+                executeStartTime: {
+                  min: {
```

Comment:
Yea, this might be a holdover from the rule-level aggregation, so it's possible to just retrieve the value from the source like we do for some of the other fields. But I don't think doing an agg over 1 doc is heavyweight tho.

Reply:
I can live with it this way - wouldn't be surprised if I found retrieving the value from source to be a little more obscure :-)